x/auth/tx/builder.go

```diff
@@ -357,3 +357,22 @@ func (w *wrapper) SetNonCriticalExtensionOptions(extOpts ...*codectypes.Any) {
 	w.tx.Body.NonCriticalExtensionOptions = extOpts
 	w.bodyBz = nil
 }
+
+func (w *wrapper) AddAuxSignerData(data tx.AuxSignerData) error {
+	err := data.ValidateBasic()
+	if err != nil {
+		return err
+	}
+
+	w.bodyBz = data.SignDoc.BodyBytes
+	w.SetTip(data.GetSignDoc().GetTip())
+
+	w.setSignerInfos(append(w.tx.AuthInfo.SignerInfos, &tx.SignerInfo{
```

Comment:
What if this is the wrong order? This method should check the signers and insert it in the correct position I think

Reply:
I added this in b9f3cf3f43f8f93bcb48731a3323e3a1c025fae5. It required 2 changes:
- adding the interface registry to the wrapper, because it's needed to decode `data.SignDoc.BodyBytes` to get the Msgs.GetSigners()
- each AuxSignerData has the signer's PubKey, but not address. We'll either need to use a bech32 converter (aka for now: the global sdk.AccAddress), or maybe we can add a new `address` field in AuxSignerData

Reply:
I'll try adding a new `address` field in AuxSignerData